fdbserver/ptxn/TLogServer.actor.cpp

```diff
@@ -1489,9 +1497,100 @@ ACTOR Future<Void> servicePeekRequest(
+			//TraceEvent("TLogPeekResults", self->dbgid).detail("ForAddress", replyPromise.getEndpoint().getPrimaryAddress()).detail("Tag1Results", s1).detail("Tag2Results", s2).detail("Tag1ResultsLim", kv1.size()).detail("Tag2ResultsLim", kv2.size()).detail("Tag1ResultsLast", kv1.size() ? kv1[0].key : "").detail("Tag2ResultsLast", kv2.size() ? kv2[0].key : "").detail("Limited", limited).detail("NextEpoch", next_pos.epoch).detail("NextSeq", next_pos.sequence).detail("NowEpoch", self->epoch()).detail("NowSeq", self->sequence.getNextSequence());
+
+			state std::vector<std::pair<IDiskQueue::location, IDiskQueue::location>> commitLocations;
+			state bool earlyEnd = false;
+			uint32_t mutationBytes = 0;
+			state uint64_t commitBytes = 0;
+			state Version firstVersion = std::numeric_limits<Version>::max();
+			for (int i = 0; i < kvrefs.size() && i < SERVER_KNOBS->TLOG_SPILL_REFERENCE_MAX_BATCHES_PER_PEEK; i++) {
+				auto& kv = kvrefs[i];
+				VectorRef<SpilledData> spilledData;
+				Version currentVersion = decodeVersionFromStorageTeamMessageRefs(kv.key);
+				versionsFromDisk.insert(currentVersion);
+				BinaryReader r(kv.value, AssumeVersion(logData->protocolVersion));
+				r >> spilledData; // here it broke
+				for (const SpilledData& sd : spilledData) {
+					if (mutationBytes >= TLOG_PEEK_REQUEST_REPLY_SIZE_CRITERIA) {
+						earlyEnd = true;
+						break;
+					}
+					if (sd.version >= req.beginVersion) {
+						firstVersion = std::min(firstVersion, sd.version);
+						const IDiskQueue::location end = sd.start.lo + sd.length;
+						commitLocations.emplace_back(sd.start, end);
+						// This isn't perfect, because we aren't accounting for page boundaries, but should be
+						// close enough.
+						commitBytes += sd.length;
+						mutationBytes += sd.mutationBytes;
+					}
+				}
+				if (earlyEnd)
+					break;
+			}
+			earlyEnd = earlyEnd || (kvrefs.size() >= SERVER_KNOBS->TLOG_SPILL_REFERENCE_MAX_BATCHES_PER_PEEK + 1);
+			wait(self->peekMemoryLimiter.take(TaskPriority::TLogSpilledPeekReply, commitBytes));
+			state FlowLock::Releaser memoryReservation(self->peekMemoryLimiter, commitBytes);
+			state std::vector<Future<Standalone<StringRef>>> messageReads;
+			messageReads.reserve(commitLocations.size());
+			for (const auto& pair : commitLocations) {
+				messageReads.push_back(
+				    tlogGroup->second->rawPersistentQueue->read(pair.first, pair.second, CheckHashes::True));
+			}
+			commitLocations.clear();
+			wait(waitForAll(messageReads));
+
+			state Version lastRefMessageVersion = 0;
+			state int index = 0;
+			loop {
+				if (index >= messageReads.size())
+					break;
+				Standalone<StringRef> queueEntryData = messageReads[index].get();
+				uint8_t valid;
+				const uint32_t length = *(uint32_t*)queueEntryData.begin();
+				queueEntryData = queueEntryData.substr(4, queueEntryData.size() - 4);
+				BinaryReader rd(queueEntryData, IncludeVersion());
+				TLogQueueEntry entry;
+				rd >> entry >> valid;
+				ASSERT(valid == 0x01);
```

Comment:
Can you define a const value for this? And use it at place when writing the entry.